spec/controllers/queued_videos_controller_spec.rb

```diff
@@ -0,0 +1,117 @@
+require 'spec_helper'
+
+describe QueuedVideosController do
+  describe 'GET index' do
+    context 'with authentication' do
+      let(:user) { Fabricate(:user) }
+      let!(:queue1) { Fabricate(:queued_video, user: user) }
+      let!(:queue2) { Fabricate(:queued_video, user: user) }
+
+      before do
+        session[:user_id] = user.id
+        get :index
+      end
+
+      it 'sets @queued_videos if user authenticated' do
+        expect(assigns(:queued_videos)).to eq([queue1, queue2])
+      end
+
+      it 'renders the index template if authenticated' do
+        expect(response).to render_template(:index)
+      end
+    end
+
+    it 'redirects to /login if not authenticated' do
+      get :index
+      expect(response).to redirect_to(login_path)
+    end
+  end
+
+  describe 'POST create' do
+    context 'with authentication' do
+      let(:user) { Fabricate(:user) }
+      let!(:video1) { Fabricate(:video) }
+
+      before do
+        session[:user_id] = user.id
+        post :create, video_id: video1
+      end
+
+      it 'creates a new queued_video' do
+        expect(QueuedVideo.count).to eq(1)
+      end
+
+      it 'creates a new queued_video with order set to max order + 1' do
+        expect(QueuedVideo.first.order).to eq(1)
+      end
+
+      it 'creates a new queued_video associated with logged in user' do
+        expect(QueuedVideo.first.user).to eq(user)
+      end
+
+      it 'creates a new queued_video associated with the video' do
+        expect(QueuedVideo.first.video).to eq(video1)
+      end
+
+      it 'does not create a new queued_video if the video is already queued for user' do
+        post :create, video_id: video1 #duplicate call of post create for this test
+        expect(QueuedVideo.count).to eq(1)
+      end
+
+      it 'redirects to queued_videos#index' do
```

Comment:
I also wouldn't hard code the controller#action as that could change. Remember that tests are a story which a new developer on the project will read. The less chance you have of the spec descriptions being wrong, the better.